ast/dml.go

```diff
@@ -983,6 +986,9 @@ func (n *SelectStmt) Restore(ctx *format.RestoreCtx) error {
 		if n.TableHints != nil && len(n.TableHints) != 0 {
 			ctx.WritePlain("/*+ ")
 			for i, tableHint := range n.TableHints {
+				if i != 0 {
+					ctx.WritePlain(", ")
+				}
```

Comment:
No, table hints *are* supposed to be separated by spaces, not commas. Demo on MySQL 8:

```
mysql> select /*+ set_var(sql_mode='x') set_var(timestamp=1) */ @@timestamp;
+-------------+
| @@timestamp |
+-------------+
|    1.000000 |
+-------------+
1 row in set, 1 warning (0.00 sec)

mysql> select /*+ set_var(sql_mode='x'), set_var(timestamp=1) */ @@timestamp;
+-------------------+
| @@timestamp       |
+-------------------+
| 1606280193.327941 |
+-------------------+
1 row in set, 2 warnings (0.00 sec)
```

Reply:
But we use commas in our [user docs](https://docs.pingcap.com/zh/tidb/stable/optimizer-hints), so we need to modify the user docs?

Reply:
better modify the docs to align with MySQL. the optional comma can be thought as TiDB's syntax extension.